contrib/pg_prewarm/pg_prewarm.c

```diff
@@ -192,6 +194,14 @@ pg_prewarm(PG_FUNCTION_ARGS)
 			Buffer		buf;
 
 			CHECK_FOR_INTERRUPTS();
+			if (block == first_block || block > last_prefetch_block) {
+				int i;
+				int prefetch_limit = Min(last_block-block, seqscan_prefetch_buffers);
+				for (i = 1; i <= prefetch_limit
+						 && PrefetchBuffer(rel, forkNumber, block + i).initiated_io;
```

Comment:
> but it does increase the number of times we try to prefetch the block we are going to process next.

Can you please elaborate this a little bit more?
Assume that we try to load page P1, and prefetching pages P2, P3, P4 (prefetch_buffers=3).
Assume that P3 is already present in shared buffers.
With the approach currently implemented i pg_prewarm, prefetch stops at P3.
Requests for P2 will not cause any calls of PrefetchRequest. And read of P3 will also trigger prefetch of P4, P5, and P6.
So for each page PrefetchBuffer is called exactly once.
How it can cause increase of number of times we try to prefetch next block?

> so you probably want to prefetch block before starting prefetching other buffers, 

But this is exactly what current done in heapam.c: target page is loaded after issuing prefetch requests.
If it cause backend to wait processing of each prefetch requests then may be it is something wrong with new logic of handling PrefetchRequests?